vividus-plugin-web-app-playwright/src/main/java/org/vividus/ui/web/playwright/BrowserContextProvider.java

```diff
@@ -17,50 +17,69 @@
 package org.vividus.ui.web.playwright;
 
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.Optional;
 
 import com.microsoft.playwright.Browser;
 import com.microsoft.playwright.BrowserContext;
 import com.microsoft.playwright.BrowserType.LaunchOptions;
 import com.microsoft.playwright.Page;
 import com.microsoft.playwright.Playwright;
+import com.microsoft.playwright.Request;
 import com.microsoft.playwright.Tracing;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.vividus.testcontext.TestContext;
+import org.vividus.ui.web.playwright.reporter.RequestsCollectingStoryReporter;
 
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 public class BrowserContextProvider
 {
+    public static final Class<Request> REQUESTS_COLLECTION_CONTEXT_KEY = Request.class;
     private static final Logger LOGGER = LoggerFactory.getLogger(BrowserContextProvider.class);
     private static final Class<BrowserContext> BROWSER_CONTEXT_KEY = BrowserContext.class;
 
     private final BrowserType browserType;
     private final LaunchOptions launchOptions;
     private final TestContext testContext;
     private final BrowserContextConfiguration browserContextConfiguration;
+    private final RequestsCollectingStoryReporter requestsCollectingStoryReporter;
 
     private PlaywrightContext playwrightContext;
 
     public BrowserContextProvider(BrowserType browserType, LaunchOptions launchOptions, TestContext testContext,
-            BrowserContextConfiguration browserContextConfiguration)
+            BrowserContextConfiguration browserContextConfiguration,
+            RequestsCollectingStoryReporter requestsCollectingStoryReporter)
```

Comment:
introduce a new interface, e.g. `NetworkRecordingStateProvider`, add one method to it `isRecordingEnabled()` and inject object of story reporter via this interface